src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -289,10 +295,20 @@ namespace cryptonote
 
       tx_out out;
       out.amount = dst_entr.amount;
-      txout_to_key tk;
-      tk.key = out_eph_public_key;
+
+      txout_to_key tk = AUTO_VAL_INIT(tk);
+
+      if (dst_entr.addr.is_swap_addr) {
+        // Zero key for swap txs
+        // TODO Should probably do all address math only when neccessary
```

Comment:
Could we do an impossible key, like `000000000000000000000000000000000000000` ?

Reply:
Not sure, for now will just implement boolberry logic. Will have some free time by the end of the week, then i will give some more precise answer.